java/dataset/src/main/cpp/jni_wrapper.cc

```diff
@@ -368,28 +368,105 @@ std::shared_ptr<arrow::Buffer> LoadArrowBufferFromByteBuffer(JNIEnv* env, jobjec
 
 inline bool ParseBool(const std::string& value) { return value == "true" ? true : false; }
 
+inline bool ParseChar(const std::string& value) {
+  if (value.size() != 1) {
+    JniThrow("Csv convert option " + value + " should be a char");
+  }
+  return value.at(0);
+}
+
 /// \brief Construct FragmentScanOptions from config map
 #ifdef ARROW_CSV
+
+bool setCsvConvertOptions(arrow::csv::ConvertOptions& options, const std::string& key,
```

Comment:
nit 

```suggestion
bool SetCsvConvertOptions(arrow::csv::ConvertOptions& options, const std::string& key,
```

Reply:
And probably, capitalize the first letter of the following method names as well.